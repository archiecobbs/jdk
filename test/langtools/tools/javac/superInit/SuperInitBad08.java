/*
 * @test /nodynamiccopyright/
 * @bug 8193760
 * @summary Test invalid reference to "this" prior to superclass construction
 *
 * @compile/fail/ref=SuperInitBad08.out -XDrawDiagnostics SuperInitBad08.java
 */
class SuperInitBad08 {

    SuperInitBad08() {
        this(this);
    }

    SuperInitBad08(Object obj) {
    }
}
