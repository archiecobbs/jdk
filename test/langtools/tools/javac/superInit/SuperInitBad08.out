SuperInitBad08.java:11:14: compiler.err.cant.ref.before.ctor.called: this
1 error
